File: modules/webrtc-core/src/main/java/com/shterneregen/securelan/webrtc/service/RtcSessionSnapshot.java
package com.shterneregen.securelan.webrtc.service;

import com.shterneregen.securelan.common.model.rtc.RtcSessionMode;
import com.shterneregen.securelan.common.model.rtc.RtcSessionState;

public record RtcSessionSnapshot(
        String sessionId,
        String localPeer,
        String remotePeer,
        RtcSessionMode mode,
        String dataChannelLabel,
        RtcSessionState state,
        String message
) {
}

File: modules/webrtc-core/src/main/kotlin/com/shterneregen/securelan/webrtc/service/RtcSessionSnapshot.kt
package com.shterneregen.securelan.webrtc.service

import com.shterneregen.securelan.common.model.rtc.RtcSessionMode
import com.shterneregen.securelan.common.model.rtc.RtcSessionState

@JvmRecord
data class RtcSessionSnapshot(
    val sessionId: String?,
    val localPeer: String?,
    val remotePeer: String?,
    val mode: RtcSessionMode?,
    val dataChannelLabel: String?,
    val state: RtcSessionState?,
    val message: String?,
)
